site/components/src/main/java/uk/nhs/hee/web/components/ListingPageComponent.java

```diff
@@ -58,12 +62,18 @@ public void doBeforeRender(final HstRequest request, final HstResponse response)
         final HstQuery query = buildQuery(request, listingPage);
         LOGGER.debug("Execute query: {}", query.getQueryAsString(false));
 
-        final HstQueryResult execute = query.execute();
+        final HstQueryResult results = query.execute();
+
+        // Removes/filters out the documents whose page is not found.
+        final List<HippoBean> resultsWithoutNonPageBeans = StreamSupport.stream(
```

Comment:
Would this be more efficient handled in the query? I am not sure wether this is possible.

Reply:
Yes, it would be. But, I already have spent few hours looking to accommodate this through query with no luck. Happy for you to suggest if there is a way to accomplish through query. Thank you!